lib/promoted/ruby/client/options.rb

```diff
@@ -166,25 +168,24 @@ def request_params include_insertion: true
         end
 
         def compact_insertions
-          @compact_insertions = []
-          insertions_to_compact = full_insertion
-          paging = request[:paging] || {}
-          size = paging[:size]
-          unless size.nil? || size == 0
-            insertions_to_compact = insertions_to_compact[0..size-1]
-          end
-          from = paging[:from].to_i
+          @insertion            = [] # insertion should be set according to the compact insertion
+          paging                = request[:paging] || {}
+          size                  = paging[:size] ? paging[:size].to_i : 0
+          from                  = paging[:from].to_i
+          insertions_to_compact = full_insertion[from..size-1]
+
           insertions_to_compact.each_with_index do |insertion_obj, index|
             # TODO - this does not look performant.
-            insertion_obj = insertion_obj.transform_keys{ |key| key.to_s.to_underscore.to_sym }
+            insertion_obj                = insertion_obj.transform_keys{ |key| key.to_s.to_underscore.to_sym }
             insertion_obj[:user_info]    = user_info
```

Comment:
Is this a copy of the original insertion?  It looks like it.  I just want to double check.

Which fields should be set in this loop?  Ones that might get set as higher-level options?

Reply:
Yes this is a copy of original insertion.